packages/web-components/fast-components/src/styles/density.ts

```diff
@@ -0,0 +1,249 @@
+    "density-component-horizontal-outer-units"
+).withDefault(3);
+
+/**
+ * The design unit multiplier for component density horizontal padding between elements.
+ *
+ * @remarks
+ * Ex: `2` to set the base padding to 2 units.
+ * May be fractional.
+ *
+ * @defaultValue 2
+ *
+ * @public
+ */
+export const densityComponentHorizontalBetweenUnits = create<number>(
+    "density-component-horizontal-between-units"
+).withDefault(2);
+
+/**
+ * The design unit multiplier for adjustments to component horizontal density tokens, which adds the specified design units to the base setting.
+ *
+ * @remarks
+ * Ex: `1` to increase padding by one unit.
+ * May be fractional.
+ *
+ * @defaultValue 0
+ *
+ * @public
+ */
+export const densityComponentHorizontalAdjustmentUnits = create<number>(
+    "density-component-horizontal-adjustment-units"
+).withDefault(0);
+
+/**
+ * The calculated cumulative value for adjustments to component horizontal density relative up the document hierarchy.
+ *
+ * @remarks
+ * To use relative density do not change this value, only consume it in styles.
+ * If you want to break the chain of relative density, override this default function with a fixed value.
+ *
+ * @public
+ */
+export const densityComponentHorizontalAdjustmentUnitsCumulative = create<number>(
+    "density-component-horizontal-adjustment-units-cumulative"
+);
+densityComponentHorizontalAdjustmentUnitsCumulative.withDefault(
+    (element: HTMLElement) => {
+        const parent = composedParent(element);
+        const valParent = parent
+            ? densityComponentHorizontalAdjustmentUnitsCumulative.getValueFor(parent)
+            : 0;
+        const valThis = densityComponentHorizontalAdjustmentUnits.getValueFor(element);
+        return valParent + valThis;
+    }
+);
+
+/**
+ * The raw calculated measurement for outer padding not accounting for border thickness.
+ *
+ * @remarks
+ * Calculated from the base `densityComponentHorizontalOuterUnits`, the adjusted `densityComponentHorizontalAdjustmentUnits` and the `designUnit` size.
+ *
+ * @public
+ */
+export const densityComponentHorizontalOuterValue = create<number>(
+    "density-component-horizontal-outer-value"
+).withDefault(
+    (element: HTMLElement) =>
+        (densityComponentHorizontalOuterUnits.getValueFor(element) +
+            densityComponentHorizontalAdjustmentUnitsCumulative.getValueFor(element)) *
+        designUnit.getValueFor(element)
+);
+
+/**
+ * The calculated measurement for outer padding adjusted for border thickness of `strokeWidth`.
+ *
+ * @remarks
+ * Calculated from `densityComponentHorizontalOuterValue` - `strokeWidth`.
+ *
+ * @public
+ */
+export const densityComponentHorizontalOuterPadding = create<number>(
+    "density-component-horizontal-outer-padding"
+).withDefault(
+    (element: HTMLElement) =>
+        densityComponentHorizontalOuterValue.getValueFor(element) -
+        strokeWidth.getValueFor(element)
```

Comment:
I went back and forth between using `strokeWidth` or a new density-specific value I had added. The problem is that the tokens generally apply at a component level, but the border stroke or padding may be applied in css at any of the child elements. Nested layout scenarios may be difficult to describe in this model, say where a border is applied to a container div, and padding isn't applied until a few levels deeper in the dom, or when there may be placeholders for invisible borders or something.

What I was really looking for was a way to set a design token value for a specific element within the styles. Like this `.control` element should have a border for the sake of density, and then I'm also going to apply density padding here.

I had a helper function that applied a property name that was _not_ a full design token, then that property was referenced in these "outer" density values. At that point they had to be `cssPartial`s though, which don't notify on update. That was causing other problems so I went back to the more traditional model.

I do think it would be useful to be able to easily scope a design token value to a particular selector within the styles, but I don't have a great example for it right now. I was considering a new Behavior to set a token value, perhaps with an optional selector to target a child element, but that starts to break the DesignToken update model as well unless the token value can be set in code rather than css.